### ISSUE TITLE
vm/cc/DartAPI_InvokeVMServiceMethod_Loop flaky crashes

### ISSUE THREAD

**Issue by rmacnak-google** [dart-lang/sdk#59573]
vm/cc/DartAPI_InvokeVMServiceMethod_Loop is flaky crashing on

vm-linux-debug-ia32
vm-reload-rollback-linux-debug-x64
vm-reload-linux-debug-x64
vm-win-release-arm64
vm-linux-release-arm64
vm-reload-linux-release-x64
vm-linux-debug-x64
vm-reload-rollback-linux-release-x64
vm-linux-release-ia32
vm-win-release-x64
vm-win-debug-x64
vm-linux-release-simarm
vm-linux-debug-simriscv64
vm-linux-release-x64

[log](https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket/8730905435054676049/+/u/test_results/ignored_flaky_test_failure_logs)

**Comment by mraleph** [comment]
Duplicate of https://github.com/dart-lang/sdk/issues/57096

**Comment by mraleph** [comment]
See also my comment here: https://dart-review.googlesource.com/c/sdk/+/395521/comment/ff4233b5_e8b2cc1d/

tldr: based on the disassembly around the crash location I think it is crashing because it invocation of `Dart_InvokeVMServiceMethod` is racing with the Dart VM shutdown. I think the actual crash might be caused by my refactoring of native ports (because I think the crash is somewhere around `Dart_CloseNativePort`).
